app/src/main/java/com/example/detodito/Management.kt

```diff
@@ -0,0 +1,168 @@
+                        }
+                    } else {
+                        dialog.dismiss()
+                        handler.post {
+                            Toast.makeText(applicationContext, "Error al crear la categoría", Toast.LENGTH_LONG).show()
+                        }
+                        Log.d("NOonResponse", response.toString())
+                    }
+                }
+            })
+        }
+
+        binding.typeSecondInputLayout.setEndIconOnClickListener {
+            val adapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, types)
+            builder.setAdapter(adapter) { _, which ->
+                val data = types[which]
+                binding.typeSecondCompleteTextView.setText(data)
+                binding.typecategoInputLayout.isEnabled = true
+                binding.typeCategoCompleteTextView.setText("")
+            }
+
+            builder.create().show()
+        }
+
+        binding.typecategoInputLayout.setEndIconOnClickListener {
+            url = "http://10.0.2.2:3000/categories"
+            val request = Request.Builder()
+                .url(url)
+                .get()
+                .build()
+
+            OkHttpClient().newCall(request).enqueue(object : Callback {
+                override fun onFailure(call: Call, e: IOException) {
+                    // Manejar errores
+                }
+
+                override fun onResponse(call: Call, response: Response) {
+                    val responseData = response.body?.string()
+                    val gson = Gson()
+                    val categories = gson.fromJson(responseData, categories::class.java)
+                    val categoriesArray = mutableListOf<categoriesItem>()
+                    for (element in categories) {
+                        categoriesArray.add(element)
+                    }
+
+                    val refresh = Handler(Looper.getMainLooper())
+                    refresh.post(kotlinx.coroutines.Runnable {
+                        val nameCategories = mutableListOf<String>()
+                        for (element in categoriesArray) {
+                            if (binding.typeSecondCompleteTextView.text.toString() == element.type) {
+                                nameCategories.add(element.name)
+                            }
+                        }
+                        val adapter = ArrayAdapter(applicationContext, android.R.layout.simple_list_item_1, nameCategories)
+                        builder.setAdapter(adapter) { _, which ->
+                            val data = nameCategories[which]
+                            binding.typeCategoCompleteTextView.setText(data)
+                        }
+
+                        builder.create().show()
+                    })
+
+                    // Aquí puedes hacer lo que necesites con el array de categorías
+                }
+            })
+        }
+    }
```

Comment:
binding.typecategoInputLayout.setEndIconOnClickListener {
    val url = "http://10.0.2.2:3000/categories"
    val request = Request.Builder()
        .url(url)
        .get()
        .build()

    OkHttpClient().newCall(request).enqueue(object : Callback {
        override fun onFailure(call: Call, e: IOException) {
            // Manejar errores de conexión o llamada a la API
            Log.e("onFailure", "Error en la llamada a la API", e)
        }

        override fun onResponse(call: Call, response: Response) {
            val responseData = response.body?.string()
            val gson = Gson()
            val categories = gson.fromJson(responseData, categories::class.java)
            val categoriesArray = mutableListOf<categoriesItem>().apply {
                addAll(categories)
            }

            val refresh = Handler(Looper.getMainLooper())
            refresh.post(kotlinx.coroutines.Runnable {
                val nameCategories = mutableListOf<String>()
                for (element in categoriesArray) {
                    if (binding.typeSecondCompleteTextView.text.toString() == element.type) {
                        nameCategories.add(element.name)
                    }
                }
                val adapter = ArrayAdapter(applicationContext, android.R.layout.simple_list_item_1, nameCategories)
                builder.setAdapter(adapter) { _, which ->
                    val data = nameCategories[which]
                    binding.typeCategoCompleteTextView.setText(data)
                }

                builder.create().show()
            })
        }
    })
}